resources/assets/coffee/react/_components/beatmap-icon.coffee

```diff
@@ -37,7 +37,9 @@ el = React.createElement
 
     showTitle = @props.showTitle && !@props.overrideVersion?
 
+    mode = if beatmap.convert then 'osu' else beatmap.mode
```

Comment:
put them last then? (also switching mode probably should select the first beatmap instead of the last one)